### ISSUE TITLE
Rule proposal: `no-useless-async`

### ISSUE THREAD

**Issue by erunion** [sindresorhus/eslint-plugin-unicorn#768]
This might be more useful in [eslint-plugin-jest](https://github.com/jest-community/eslint-plugin-jest) where something like this is more common, but it would be nice to have a rule that could catch async function that don't need to be async.

## Fail
```js
it('should expect true', async () => {
  expect(methodThatReturnsTrue()).toBe(true);
});
```

## Pass
```js
it('should expect true', () => {
  expect(methodThatReturnsTrue()).toBe(true);
});

it('should create a user', async () => {
  await user.save();

  expect(methodThatReturnsTrue()).toBe(true);
});
```

**Comment by sindresorhus** [comment]
ESLint already has a rule for this: https://eslint.org/docs/rules/require-await

**Comment by erunion** [comment]
So it does!
